src/tools/rados/rados.cc

```diff
@@ -1693,6 +1693,8 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       }
       catch (const std::runtime_error& e) {
 	cerr << e.what() << std::endl;
+	if (!stdout)
```

Comment:
This should probably be use_stdout instead.  This means that line 1709 of this file is an existing bug.

The impact of these issues are almost nothing because rados_tool_common() is called only by main() which just returns (exits) with the results.  Maybe it would be complained about by a code analysis tool like Coverity.